src/backend/adapters/AuthAdapter.ts

```diff
@@ -0,0 +1,18 @@
+import bcrypt from "bcrypt"
+import { AuthAdapter } from "../services/UserService"
+import { tryCatch } from "../utils/tryCatch"
+
+export const makeAuthAdapter = (): AuthAdapter => {
```

Comment:
well... it's as if I'm saying "I don't care how you salt the password, I just need you (AuthAdapter) to salt the password, and verify the salted password".   Will I ever need to replace how I salt?  Probably not.  But if I did it would not interfere with the larger UserService, it'd just be a new AuthAdapter that'd implement hash & verify functions.